src/main/java/tech/pegasys/tools/epchecks/DoNotReturnNullOptionals.java

```diff
@@ -13,56 +13,68 @@
 package tech.pegasys.tools.epchecks;
 
 import static com.google.errorprone.BugPattern.SeverityLevel.SUGGESTION;
-import static com.google.errorprone.matchers.Matchers.contains;
+import static com.google.errorprone.matchers.Description.NO_MATCH;
 import static com.sun.source.tree.Tree.Kind.NULL_LITERAL;
 
 import com.google.auto.service.AutoService;
 import com.google.errorprone.BugPattern;
 import com.google.errorprone.VisitorState;
 import com.google.errorprone.bugpatterns.BugChecker;
-import com.google.errorprone.bugpatterns.BugChecker.MethodTreeMatcher;
+import com.google.errorprone.bugpatterns.BugChecker.ReturnTreeMatcher;
 import com.google.errorprone.matchers.Description;
 import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.google.errorprone.util.ASTHelpers;
 import com.sun.source.tree.MethodTree;
 import com.sun.source.tree.ReturnTree;
 import com.sun.source.tree.Tree;
+import com.sun.tools.javac.code.Type;
 
-/*
- * This is reworked from an example found at:
- * https://github.com/google/error-prone/wiki/Writing-a-check
- */
-
-@AutoService(BugChecker.class) // the service descriptor
+@AutoService(BugChecker.class)
 @BugPattern(
     name = "DoNotReturnNullOptionals",
     summary = "Do not return null optionals.",
     severity = SUGGESTION,
     linkType = BugPattern.LinkType.NONE)
-public class DoNotReturnNullOptionals extends BugChecker implements MethodTreeMatcher {
+public class DoNotReturnNullOptionals extends BugChecker implements ReturnTreeMatcher {
+  private static final Matcher<Tree> IS_NULL_LITERAL = Matchers.kindIs(NULL_LITERAL);
 
-  private static class ReturnNullMatcher implements Matcher<Tree> {
+  @Override
+  public Description matchReturn(final ReturnTree tree, final VisitorState state) {
+    // Filter out statements that do not return null.
+    if (tree == null || !IS_NULL_LITERAL.matches(tree.getExpression(), state)) {
```

Comment:
hmm, so this check would only flag literal `return null` but wouldn't pick up:

```java
final String value = someBoolean ? "present" : null;
return value;
```

I'm not sure if the built-in check would handle that or not though as it does require some fairly fancy inspection logic...